src/broadcast/broadcast.rs

```diff
@@ -462,15 +454,13 @@ fn decode_from_shards(
         None // The proposer is faulty.
     } else {
         // Reconstruct the value from the data shards.
-        glue_shards(mtree, data_shard_num)
+        glue_shards(mtree, coding.data_shard_count())
     }
 }
 
-/// Concatenates the first `n` leaf values of a Merkle tree `m` in one value of
-/// type `T`. This is useful for reconstructing the data value held in the tree
-/// and forgetting the leaves that contain parity information.
+/// Concatenates the first `n` leaf values of a Merkle tree. The first four bytes are interpreted
```

Comment:
"a Merkle tree" without the name may sound like an arbitrary Merkle tree.

Reply:
You mean it should be "the Merkle tree"

Reply:
I amended:

```diff
-/// Concatenates the first `n` leaf values of a Merkle tree. The first four bytes are interpreted
-/// as the payload size, and the padding beyond that size is dropped.
+/// Concatenates the first `n` leaf values of the Merkle tree `m`. The first four bytes are
+/// interpreted as the payload size, and the padding beyond that size is dropped.
```

Reply:
Sorry, I think it should be an [indefinite description](https://plato.stanford.edu/entries/descriptions/): "a Merkle tree `m`". It might be a definite description ("the 'm'") but I'm lazy to follow the dictionary article argument. :books: :books: :books:

Reply:
OK, I amended it, too.